controllers/topology/reconcile_state.go

```diff
@@ -374,10 +385,63 @@ func (r *ClusterReconciler) reconcileReferencedTemplate(ctx context.Context, in
 	// Set up a cleanup func for removing the old template.
 	// NOTE: This function must be called after updating the object containing the reference to the Template.
 	return func() error {
-		log.Info("Deleting", in.desired.GroupVersionKind().String(), in.desired.GetName())
+		logReconcileAction(log, deleteObjectReconcileOperation, in.current)
 		if err := r.Client.Delete(ctx, in.current); err != nil {
 			return errors.Wrapf(err, "failed to delete %s/%s", in.desired.GroupVersionKind(), in.desired.GetName())
 		}
 		return nil
 	}, nil
 }
+
+type reconcileAction int
+
+const (
+	createObjectReconcileOperation reconcileAction = iota
```

Comment:
```suggestion
	createObjectReconcileAction reconcileAction = iota
```
nit. I think we should keep it consistent